app/components/viewable_champ/section_component.rb

```diff
@@ -0,0 +1,45 @@
+class ViewableChamp::SectionComponent < ApplicationComponent
+  include ApplicationHelper
+  include TreeableConcern
+
+  def initialize(champs: nil, nodes: nil, demande_seen_at:, profile:)
+    @demande_seen_at, @profile, @repetition = demande_seen_at, profile
+    if nodes.nil?
+      nodes = to_tree(champs:)
+    end
+    @nodes = to_sections(nodes:)
+  end
+
+  def section_id
+    @section_id ||= header_section ? dom_id(header_section, :content) : SecureRandom.uuid
+  end
+
+  def header_section
+    return @nodes.first if @nodes.first.is_a?(Champs::HeaderSectionChamp)
```

Comment:
un peu dur à lire d'avoir un return sans rien en dessous :  je propose

```suggestion
    if @nodes.first.header_section?  # si possible, sinon :
      # if @nodes.first.is_a?(Champs::HeaderSectionChamp)
      @nodes.first
    end
```

Reply:
pas possible, parmis les noeuds, il peut y avoir une section.:/